packages/manager/modules/nasha/src/components/partition/snapshot/delete/delete.controller.js

```diff
@@ -1,25 +1,21 @@
 export default class NashaComponentsPartitionSnapshotDeleteController {
   /* @ngInject */
-  constructor($translate, $http) {
-    this.$translate = $translate;
+  constructor($http) {
     this.$http = $http;
   }
 
   submit() {
-    const { serviceName } = this.nasha;
-    const { partitionName } = this.partition;
     const { customSnapshotName } = this;
 
-    this.$http
-      .delete(
-        `/dedicated/nasha/${serviceName}/partition/${partitionName}/customSnapshot/${customSnapshotName}`,
-      )
-      .then(({ data: { taskId } }) =>
+    return this.$http
+      .delete(`${this.partitionApiUrl}/customSnapshot/${customSnapshotName}`)
+      .catch((error) => this.close({ error }))
+      .then(({ data: task }) =>
```

Comment:
Doing that, it will probably introduce a mistake in HTTP error case.
For this case, the catch function will be executed and also the then function.

The issue will come into the then function, because we don't have a task in error case.
maybe implement then and catch after.

Let me know.

Reply:
Sure !
There are others places where I did the same mistake
This has already been taken care of in this PR : https://github.com/ovh/manager/pull/7494/files#diff-3e4b347490f8c2ca396f30c26909f2342ad5f167da17def31f9d689dbd03f911